### ISSUE TITLE
Issue with Faker module

### ISSUE THREAD

**Issue by arossert** [pyinstaller/pyinstaller#3839]
I'm trying to use pyinstaller with Faker module but it always seems to be missing some imports.
I know that Faker make a lot fo dynamic imports and its a bit problematic with pyinstaller.

I saw this PR #3465 but it was never been merged.

Firs I have tired with no configuration and there were some missing modules so I have tried the "hook" that was described [here](http://pyinstaller.47505.x6.nabble.com/Problem-with-a-module-called-Faker-generates-fake-Data-td1967.html) but I'm still getting errors:
```AttributeError: Invalid configuration for faker locale `en_US````

It seems that there is a list of locales that are created dynamically and in pyinstaller binary is always empty.

What I can do to make it work?

**Comment by htgoebel** [comment]
Does #3465 solve your issue? [How to apply a Patch or Pull Request to PyInstaller](https://github.com/pyinstaller/pyinstaller/wiki/How-to-apply-a-Patch-or-Pull-Request-to-PyInstaller)

**Comment by evilcatmomo** [comment]
Could  you pleas explain how to apply it in this case ?

**Comment by htgoebel** [comment]
@evilcatmomo You are a developer, arn't you? So you should be able to transfer [How to apply a Patch or Pull Request to PyInstaller](https://github.com/pyinstaller/pyinstaller/wiki/How-to-apply-a-Patch-or-Pull-Request-to-PyInstaller) to this situation.

**Comment by arossert** [comment]
@htgoebel without using the hook I'm getting the message:
`ImportError: No module named providers`

After applying the hook from #3465 this is the message that I'm getting:
`AttributeError: Invalid configuration for faker locale `en_US``

I have added this to the code:
`from faker.config import AVAILABLE_LOCALES
print AVAILABLE_LOCALES`

and the output is:
```
(faker_test) Amirs-MacBook-Pro-2:faker_test arossert$ dist/faker_test
set([])
Traceback (most recent call last):
  File "faker_test.py", line 5, in <module>
    fake = Faker()
  File "site-packages/faker/factory.py", line 45, in create
AttributeError: Invalid configuration for faker locale `en_US`
[77677] Failed to execute script faker_test
```

It seems that the `AVAILABLE_LOCALES` are loaded dynamically and pyinstaller can't handle it, is there something that I can do in the hook to fix this?
Will be happy to provide a PR once this works.

**Comment by arossert** [comment]
I have created a fix on the "Faker" module that was merged today.
[#927](https://github.com/joke2k/faker/pull/927)
So the next release will support freezing with PyInstaller.

This hook should work with the new fix in "Faker" #4133
